api/comments_api.py

```diff
@@ -72,18 +80,19 @@ def do_get(self, **kwargs) -> dict[str, list[dict[str, Any]]]:
       lambda c: self._should_show_comment(c, user_email, is_admin), comments))
 
     dicts = [activity_to_json_dict(c) for c in comments]
-    return {'comments': dicts}
+    return GetCommentsResponse(comments=dicts).to_dict()
```

Comment:
#4213 has merged, so I think that this can return the `GetCommentResponse` directly without calling to_dict().

Reply:
The dict conversion does not work and still gives such errors when removing it `======================================================================
ERROR: test_patch__undelete_comment (api.comments_api_test.CommentsAPITest.test_patch__undelete_comment)
Handler unmarks a comment as deleted as requested by authorized user.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mark/chromium-dashboard/api/comments_api_test.py", line 249, in test_patch__undelete_comment
    self.assertEqual(get_resp['comments'][0]['deleted_by'], None)
                     ~~~~~~~~^^^^^^^^^^^^
TypeError: 'GetCommentsResponse' object is not subscriptable

======================================================================
ERROR: test_post__comment_only (api.comments_api_test.CommentsAPITest.test_post__comment_only)
Handler adds a comment only, does not require approval permission.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/unittest/mock.py", line 1375, in patched
    return func(*newargs, **newkeywargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mark/chromium-dashboard/api/comments_api_test.py", line 268, in test_post__comment_only
    self.assertEqual(actual, {'message': 'Done'})
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/unittest/case.py", line 873, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/unittest/case.py", line 863, in _baseAssertEqual
    if not first == second:
           ^^^^^^^^^^^^^^^
  File "/Users/mark/chromium-dashboard/cs-env/lib/python3.11/site-packages/chromestatus_openapi/models/base_model.py", line 64, in __eq__
    return self.__dict__ == other.__dict__
                            ^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute '__dict__'` It looks like in the test file it still expect the response to be returned by the do_get method defined in the specific handler

Reply:
Sure, the tests would need to be updated to validate the returned model object instead of a returned dict.  It's OK with me if that is left for a future PR.